core/include/usFramework.h

```diff
@@ -0,0 +1,155 @@
+/*=============================================================================
+
+  Library: CppMicroServices
+
+  Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+  file at the top-level directory of this distribution and at
+  https://github.com/saschazelzer/CppMicroServices/COPYRIGHT .
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+=============================================================================*/
+
+#ifndef USFRAMEWORK_H
+#define USFRAMEWORK_H
+
+#include <usCoreConfig.h>
+#include "usModule.h"
+
+#include <map>
+#include <string>
+
+US_BEGIN_NAMESPACE
+
+class FrameworkPrivate;
+
+/**
+ * \ingroup MicroServices
+ *
+ * The CppMicroServices Framework.
+ *
+ * <p>
+ * A Framework is itself a bundle and is known as the "System Bundle".
+ * The System Bundle differs from other bundles in the following ways:
+ * - The system bundle is always assigned a bundle identifier of zero (0).
+ * - The system bundle <code>GetLocation</code> method returns the string: "System Bundle".
+ * - The system bundle's life cycle cannot be managed like normal bundles. Its life cycle methods
+ *   must behave as follows:
+ *   - Start - Does nothing because the system bundle is already started.
+ *   - Stop - Stops all installed bundles.
+ *   - Uninstall - The Framework must throw a std::runtime_error exception indicating that the 
+ *     system bundle cannot be uninstalled.
+ *
+ * Framework instances are created using a FrameworkFactory. The methods of this class can be
+ * used to manage and control the created framework instance.
+ *
+ * @remarks This class is thread-safe.
```

Comment:
Such remarks are all over the place and they are true if the related build option is enabled. Otherwise they are certainly not. We could just add a general remark to the documentation of the build option.